DigitalLearningSolutions.Data/Services/SectionContentService.cs

```diff
@@ -27,39 +27,50 @@ public SectionContentService(IDbConnection connection, ILogger<SectionContentSer
-                                    AND NextSectionsTutorials.ArchivedDate IS NULL
-                                LEFT JOIN Sections AS NextSections
-                                    ON NextSectionsTutorials.SectionID = NextSections.SectionID
-                                    AND CurrentSection.SectionNumber <= NextSections.SectionNumber
-                                    AND (CurrentSection.SectionNumber < NextSections.SectionNumber
-                                        OR CurrentSection.SectionID < NextSections.SectionID)
-                        WHERE CurrentSection.SectionId = @sectionId
-                            AND NextSections.SectionID IS NOT NULL
-                            AND NextSections.SectionNumber IS NOT NULL
+                                INNER JOIN CustomisationTutorials AS OtherCustomisationTutorials
+                                    ON OtherCustomisationTutorials.CustomisationID = @customisationId
+                                INNER JOIN Customisations
+                                    ON OtherCustomisationTutorials.CustomisationID = Customisations.CustomisationID
+                                INNER JOIN Tutorials AS OtherSectionsTutorials
+                                    ON OtherCustomisationTutorials.TutorialID = OtherSectionsTutorials.TutorialID
+                                    AND OtherSectionsTutorials.ArchivedDate IS NULL
+                                INNER JOIN Sections AS OtherSections
+                                    ON OtherSectionsTutorials.SectionID = OtherSections.SectionID
+                                    AND CurrentSection.SectionID <> OtherSections.SectionID
+                        WHERE CurrentSection.SectionID = @sectionId
                             AND Customisations.Active = 1
-                            AND NextSections.ArchivedDate IS NULL
+                            AND OtherSections.ArchivedDate IS NULL
                             AND (
-                                 NextCustomisationTutorials.Status = 1
-                                 OR NextCustomisationTutorials.DiagStatus = 1
-                                 OR (Customisations.IsAssessed = 1 AND NextSections.PLAssessPath IS NOT NULL)
+                                 OtherCustomisationTutorials.Status = 1
+                                 OR (OtherCustomisationTutorials.DiagStatus = 1 AND OtherSections.DiagAssessPath IS NOT NULL)
+                                 OR (Customisations.IsAssessed = 1 AND OtherSections.PLAssessPath IS NOT NULL)
                             )
-                        GROUP BY
-                            CurrentSection.SectionID,
-                            NextSections.SectionID,
-                            NextSections.SectionNumber
-                        ORDER BY NextSections.SectionNumber, NextSections.SectionID
+                    ),
+                    OtherSectionsExistTable AS (
```

Comment:
Does this need to be defined as a new select? Can't the `CASE` part just be part of the select statement below? 🙂

Reply:
Yep! Done in 2137e08 🙂